cmd/juju/action/exec.go

```diff
@@ -258,3 +270,47 @@ func (c *execCommand) Run(ctx *cmd.Context) error {
 
 	return c.operationResults(ctx, &runResults)
 }
+
+// printExecOutput is the default "plain" formatter for the exec command.
+func (c *execCommand) printExecOutput(w io.Writer, value interface{}) error {
+	info, ok := value.(map[string]interface{})
+	if !ok {
+		return errors.Errorf("expected value of type %T, got %T", info, value)
+	}
+
+	// unit/machine name -> stdout+stderr
+	var outputs = make(map[string]string)
+	// Store all unit/machine names. We need to sort them before printing to
+	// ensure a consistent iteration order.
+	names := make([]string, 0, len(info))
+
+	for k := range info {
+		names = append(names, k)
+		resultMetadata, ok := info[k].(map[string]interface{})
+		if !ok {
+			return errors.Errorf("expected value of type %T, got %T", resultMetadata, info[k])
+		}
+		resultData, ok := resultMetadata["results"].(map[string]interface{})
+		if ok {
+			if stdout, ok := resultData["stdout"]; ok {
+				outputs[k] += fmt.Sprintf("%s\n", stdout)
+			}
+			if stderr, ok := resultData["stderr"]; ok {
+				outputs[k] += fmt.Sprintf("%s\n", stderr)
+			}
+		}
+	}
+
+	// Iteration order for maps is not guaranteed -> need to sort the keys first
+	sort.Strings(names)
```

Comment:
This is non-ideal if the unit number is >9, because 11 will sort before 2. Can we instead create a slice of `struct { name string; number int }` or similar, and sort by name then number?

Looks like we can us https://github.com/juju/naturalsort for this.